### ISSUE TITLE
Tests fail under ruby 2.5.0

### ISSUE THREAD

**Issue by andriytyurnikov** [rmosolgo/graphql-ruby#1295]
Failing test:
```
Failure:
GraphQL::InternalRepresentation::Rewrite::building a tree over concrete types with fragments#test_0001_groups selections by object types which they apply to [/Users/andriytyurnikov/Code/andriytyurnikov/graphql-ruby/spec/graphql/internal_representation/rewrite_spec.rb:127]
Minitest::Assertion: --- expected
+++ actual
@@ -1 +1 @@
-3
+1
```

**Comment by rmosolgo** [comment]
🤔 weird! I wonder why that would be different on 2.5 ...

**Comment by dblock** [comment]
This isn't actually caused by ruby 2.5.0, it's reproducible on all versions. There's something order-dependent. I had opened https://github.com/rmosolgo/graphql-ruby/issues/1765, which turns out is a dup - feel free to edit this one and close the newer issue or close this one.

**Comment by dblock** [comment]
I can confirm this is ruby-version dependent, will try to fix in #1764.

**Comment by rmosolgo** [comment]
FWIW the solution is here: https://github.com/rmosolgo/graphql-ruby/pull/1352 But, the question is whether slowing down `hash` is worth it, i'm not sure.
